ci/cd: fix codesigning on mac/arm cloud agent

vietage opened the pull request:
TeamCity build fails on Mac arm cloud agent because codesigning hangs on a password prompt in GUI. This commit changes how we generate a certificate and how we import it into a keychain so that it does not prompt passwords in GUI.

aarzilli commented:
The tests will pass like this because the certificate is only necessary for the macOS native backend, which we don't test (it's been broken for a long time). However if someone ever were to fix it the certificate would need to be specifically installed in the system keychain, so changing  gencert.sh to install the certificate somewhere else is just breaking the backend further.

I think we should just add something like this to checkCert in _scripts/make.go:

```
if os.Getenv("CI") != "" {
        return true
}
```

the rest probably works.

vietage commented:
Yes, it is way easier to disable codesigning at all if you do not need it.

We still need somehow pass CI=true to the pipeline. I've modified TC config and added CI=true for Mac build, but it is tricky to test because we need these changes in VCS in order for TeamCity to apply them. So I only tested this approach with custom build with manually provided CI=true env variable.

And looks like Linux/Amd64 build is flaky and sometimes fails with"cannot find git" message. I've re-run the build again and it passed.

aarzilli reviewed (approved):
LGTM
